cli/command/context/import.go

```diff
@@ -28,10 +29,19 @@ func RunImport(dockerCli command.Cli, name string, source string) error {
 	if err := checkContextNameForCreation(dockerCli.ContextStore(), name); err != nil {
 		return err
 	}
+
 	var reader io.Reader
+	var importType store.ImportType
 	if source == "-" {
 		reader = dockerCli.In()
+		importType = store.Cli
 	} else {
+		if strings.HasSuffix(source, ".zip") {
```

Comment:
I left a similar comment on https://github.com/docker/cli/pull/1867/files#r282963749, but wondering if instead we could read the first few bytes from the file to determine the compression from its header (instead of using the file-extension?)

Reply:
There is a way to identify the content of a file using `http.DetectContentType`. Do you think that will land itself better for this purpose? I wasn't sure brining in a dependency from `http` package in.

Reply:
Hm, looks like that is really targeted at downloading from a remote location; https://golang.org/src/net/http/sniff.go?s=646:688#L11

If we know the "magic" header for zips perhaps we can just read some bytes and check for that (but haven't looked into it deeply); @ijc @simonferquel any ideas?

Reply:
I have looked into this a bit more, and was not able to come up with a straightforward way to determine just using file headers, especially for `.zip`. Since this is a very common way of doing this I say let's go with it, but please let me know if there's issues with having a check like this in the grand scheme of things that I might be missing.

@ijc @simonferquel — Please let me know if you got something for handling this in another way, would love to change it to that.

Reply:
I will resolve this for now, happy to re-open if we want to pursue this further.